codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerRuntimeTypesReExportsGenerator.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.rust.codegen.server.smithy.generators
+
+import software.amazon.smithy.rust.codegen.core.rustlang.RustWriter
+import software.amazon.smithy.rust.codegen.core.rustlang.asType
+import software.amazon.smithy.rust.codegen.core.rustlang.rustTemplate
+import software.amazon.smithy.rust.codegen.core.smithy.CodegenContext
+import software.amazon.smithy.rust.codegen.server.smithy.ServerCargoDependency
+import software.amazon.smithy.rust.codegen.server.smithy.ServerRuntimeType
+
+class ServerRuntimeTypesReExportsGenerator(
+    codegenContext: CodegenContext,
+) {
+    private val runtimeConfig = codegenContext.runtimeConfig
+    private val codegenScope = arrayOf(
+        "Router" to ServerRuntimeType.Router(runtimeConfig),
+        "SmithyHttpServer" to ServerCargoDependency.SmithyHttpServer(runtimeConfig).asType(),
+    )
+
+    fun render(writer: RustWriter) {
+        writer.rustTemplate(
+            """
+            pub use #{SmithyHttpServer}::Extension;
+            pub use #{Router};
+            pub use #{SmithyHttpServer}::operation::OperationShape;
+            pub use #{SmithyHttpServer}::operation::Operation;
+
+            ##[doc(hidden)]
+            pub use #{SmithyHttpServer}::instrumentation;
+            ##[doc(hidden)]
+            pub use #{SmithyHttpServer}::proto;
```

Comment:
What's the point of doing a #[doc(hidden)] re-export? I might be missing something obvious 🤔

Reply:
It means that https://github.com/awslabs/smithy-rs/pull/1886 would only have to remove the `#[doc(hidden)]` rather than making decisions on what to include in this module or not.

This is a non-issue if we do https://github.com/awslabs/smithy-rs/pull/1909#discussion_r1005782072 though.

Reply:
> If #[doc(hidden)] is transitive and re-exporting something #[doc(hidden)] means the re-export is also hidden then this isn't necessary - I'm unsure if this is the case or not.

This was my expectation, hence the question.
I tested it and I found that:
- Re-exporting a `#[doc(hidden)]` module gives you a `#[doc(hidden)]` module, so no need to mark the rexport as `#[doc(hidden)]`;
- Any `#[doc(hidden)]` item inside the re-exported module remains `#[doc(hidden)]`.

Reply:
Code:
```rust
pub use dep::hello as hey;
pub use dep::my_hidden_module as hey_from_hidden;
```
Dependency code:
```rust
pub mod hello {
    #[doc(hidden)]
    struct MyHiddenStruct;
}

#[doc(hidden)]
pub mod my_hidden_module {
    struct MyVisibleStruct;
}
```
Screenshot of the docs:
<img width="324" alt="image" src="https://user-images.githubusercontent.com/20745048/198246412-d52a8427-2bb8-42e1-bbe7-e7a5cbd16684.png">
<img width="371" alt="image" src="https://user-images.githubusercontent.com/20745048/198246449-fee95bce-95b5-49c5-b295-6627b4d6cdf2.png">

Reply:
Good to know, we can drop the `#[doc(hidden)]` from these.